### PR TITLE
buildRustPackage: add assert for verifyCargoDeps

### PR DESCRIPTION
**Pull request by zowoq**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc @bhipple @Mic92

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 113053` at eaf55f33 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/3dbe469b67643865f9817211fb3a2490">evscript</a></li></li>
    <li><a href="https://gist.github.com/a193fbe878998185c99c4ebca89dcb03">netease-music-tui</a></li></li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 113053` at eaf55f33 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/6e7d67f4e9d4216136ad5f25dd83d539">evscript</a></li></li>
    <li><a href="https://gist.github.com/b4798cdf7297817629f25c8fdcaa458b">netease-music-tui</a></li></li>
  </ul>
</details>

**Comment by Vonfry** [comment]
netease-music-tui: LGTM

File: pkgs/build-support/rust/default.nix [hunk 1]
```diff
@@ -51,11 +51,15 @@
...
 # Otherwise, everything from the tarball would've been built/tested.
 , buildAndTestSubdir ? null
+
+, verifyCargoDeps ? ""
 , ... } @ args:
 
 assert cargoVendorDir == null -> !(cargoSha256 == "" && cargoHash == "");
 assert buildType == "release" || buildType == "debug";
 
+assert verifyCargoDeps != "" -> throw "`verifyCargoDeps` is not needed as it is the default";
```
Comment (Mic92) [inline pkgs/build-support/rust/default.nix:61]: Does this still allow to set it to false?

Comment (zowoq) [reply pkgs/build-support/rust/default.nix:61]: No? Not sure if I'm misunderstanding what you're asking but `verifyCargoDeps` is a no op anyway?

**Comment by zowoq** [comment]
Pushed the commits that remove `verifyCargoDeps` to master, we add the assert if it sneaks back in again.

**Pull request closed without merging**